sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -170,6 +169,13 @@ def _get_url_for_timerange(self, timerange):
             result.append(url)
         return result
 
+    def post_search_hook(self, exdict, matchdict):
+        # update the extracted metadata to include the queried times rather
```

Comment:
Is this going to add the correct row time, not the whole query time right? i.e. if I search for a week in a continuous time range I will get 7 one day results and files correct?

Reply:
it will return 7 results with 7 files. The time in the UnifiedResponse table will still say the beginning to start of the year - but the files returned will be now the ones of interest.

e.g. 

```python
>>> res = Fido.search(a.Time('2013-10-07', '2013-10-13'), a.Instrument.srs_table)

<sunpy.net.fido_factory.UnifiedResponse object at 0x7ffc98f3a2d0>
Results from 1 Provider:

7 Results from the SRSClient:
     Start Time           End Time      Instrument Physobs Source Provider
------------------- ------------------- ---------- ------- ------ --------
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA
2013-01-01 00:00:00 2013-12-31 23:59:59  SRS-TABLE     SRS   SWPC     NOAA

>>> files = Fido.fetch(res)

<parfive.results.Results object at 0x7ffc97796610>
['/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt', '/Users/laurahayes/sunpy/data/20131007SRS.txt']
```

Reply:
oh wait hold on I think something else i wrong

Reply:
I would expect that if this works, all of those times will be the days they expect to be.

Reply:
I'm going to request changes on this so it doesn't get merged before this is resolved.

Reply:
Has this been resolved?

Reply:
yep! as far as I understand @Cadair's point, but it all the client works as expected now